render-app/src/main/java/org/janelia/alignment/spec/stack/StackIdNamingGroup.java

```diff
@@ -0,0 +1,48 @@
+package org.janelia.alignment.spec.stack;
+
+import java.io.Serializable;
+
+/**
+ * Specifies naming patterns for a group of stacks produced by a common processing step
+ * (e.g. alignment or intensity correction).
+ *
+ * @author Eric Trautman
+ */
+public class StackIdNamingGroup
```

Comment:
I agree the class name is poor and is likely the source for some confusion.  I'll try to explain the purpose here and maybe a better name will emerge ...  Organizationally, render stacks each have an owner, a project, and a base name (stack).  The three components together are considered a StackId.  For pipelines, we potentially need to process multiple stackIds for the same owner.  The projectPattern allows for specification of multiple project names and the stackPattern allows for specification of multiple stack names.  The NamingGroup allows for specification of the patterns in the context of a group of stacks produced by a related process (e.g. raw imaging, alignment, or intensity correction).

Reply:
Thanks, I think I understand the intent better now.

Still, it feels to me as if the group should be a collection of `StackId`s rather than membership tests (the fact that these membership tests are publicly exposed creates most confusion for me). Is there a way of only using the patterns in a static creation method, e.g., `StackIdGroup.from(String projectPattern, String stackPattern, [RenderWebService])`?

Reply:
One goal I had was to limit/optimize compilation of the patterns so that compilation would only occur once - before looping through all of the stacks that needed to be matched.  Because the NamingGroups are optional and are combined with default patterns from the StackIdWithZParameter object, it made sense to put the matching logic there rather than in the NamingGroup object (which might not even exist).